src/compressed_tensors/quantization/lifecycle/initialize.py

```diff
@@ -58,7 +61,12 @@ def initialize_module_for_quantization(
         _initialize_scale_zero_point_observer(module, "input", scheme.input_activations)
     if scheme.weights is not None:
         if hasattr(module, "weight"):
-            _initialize_scale_zero_point_observer(module, "weight", scheme.weights)
+            weight_shape = None
+            if isinstance(module, torch.nn.Linear):
```

Comment:
Does this limit us to only quantizing the Linear layer??

Reply:
no, should just for having the shape to preallocate the scale/zp for. the `_initialize...` call is outside this if